design/ceph/ceph-exporter.md

```diff
@@ -0,0 +1,121 @@
+# Ceph exporter
+
+### Goals
+Use Ceph exporter as the new source of metrics based on performance counters coming from every Ceph daemon.
+
+## Use case
+Ceph exporter was created to avoid performance problems at scale obtaining metrics in clusters with high number of daemons (mainly OSDs).
+The Prometheus manager module was not enough performant to obtain the performance counters from all the daemons along the whole Ceph cluster, so the solution for avoid this problem in big clusters is to have a new daemon collecting all the performance counters from all the daemons in each host, and expose them as standard prometheus metrics at /metrics endpoint for prometheus to scrape each of those endpoints.
+### Why do we need this daemon in Rook
+Although probably the Rook clusters usually do not have a huge number of daemons, and therefore, probably the main problem solved by the Ceph Exporter(scalability) is not going to be present, with the Ceph exporter, the source of metrics changes in the Ceph clusters.
```

Comment:
Is the central mgr module still going to exist? Or the new ceph exporter daemon will be required? If both are options, seems like we might want to have an option for smaller clusters to still report to the mgr, and larger clusters to report to the ceph exporter. But if the only supported option will be the exporter, then we will always need to start it.

We do have many large clusters in rook upstream, so let's not assume rook doesn't benefit from the change. 
```suggestion
With the Ceph exporter, the source of metrics changes in the Ceph clusters.
```

Reply:
Yes, we still need prometheus module exporter for cluster-wide metrics. Well, at least in latest quincy we still haven't removed the perf counters metrics support from prometheus module, but we have ceph-exporter too to do this job. 
After this release we're removing perf counters from module and only exporter will take care of this. But we've discussed to introduce a module option for prometheus module to enable/disable support for fetching perf counters metrics (`default: false`)

Reply:
Additionally, `ceph-exporter` should be deployed by default in rook cluster (recommended). The module option will be useful in case if exporter pods failed to start or any issues with container not running then user still can have option to get metrics from module.